apps/designer/app/designer/shared/fonts-manager/fonts-manager.tsx

```diff
@@ -80,62 +92,116 @@ export const FontsManager = ({ value, onChange }: FontsManagerProps) => {
   };
 
   const fontItems = useMemo(() => getItems(assets), [assets]);
+
+  const [openMenuItem, setOpenMenuItem] = useState<Item>();
+
   const selectedItem =
-    fontItems.find((item) => item.label === value) ?? fontItems[0];
-
-  const { items, getComboboxProps, getMenuProps, getItemProps, getInputProps } =
-    useCombobox({
-      items: fontItems,
-      value: selectedItem,
-      itemToString: (item) => item?.label ?? "",
-      onItemSelect: (value) => {
-        if (value !== null) {
-          onChange(value.label);
-        }
-      },
-    });
-
-  const { isEmpty, ...menuProps } = getMenuProps();
+    // After deletion the selected item may not be in the list anymore.
+    fontItems.find((item) => item.label === value) ??
+    fontItems.find((item) => item.type !== "category") ??
+    null;
+
+  const {
+    items: filteredItems,
+    resetFilter,
+    ...comboboxProps
+  } = useCombobox({
+    items: fontItems,
+    value: selectedItem,
+    itemToString(item) {
+      return item?.type === "category" ? "" : item?.label ?? "";
+    },
+    onItemSelect(value) {
+      if (value !== null) {
+        onChange(value.label);
+      }
+    },
+  });
+
+  const items = groupItems(filteredItems);
+
+  useEffect(resetFilter, [fontItems.length, resetFilter]);
```

Comment:
Is the intention to run `resetFilter` when `fontItems.length` changes? Should `resetFilter` be a dependency?

Also, React does not guarantee that it will run effect only when dependencies change, they are just a hint for performance. Although, I couldn't find this in their current docs, so maybe they've gave up and guarantee it now 😄